activerecord/lib/active_record/enum.rb

```diff
@@ -138,19 +152,26 @@ def enum(definitions)
         _enum_methods_module.module_eval do
           pairs = values.respond_to?(:each_pair) ? values.each_pair : values.each_with_index
           pairs.each do |value, i|
+            value_prefix = \
```

Comment:
Why not just do:

```
if prefix == true
  prefix = name
end

if prefix
  method_name = "#{prefix}_#{value}"
else
  method_name = value
end
```

I'd rather avoid adding ternaries for something like this